### PR TITLE
Improve the documentation for `energy_edges`

### PR DESCRIPTION
**Pull request by Astro-Kirsty**
**Description**
This PR is to fix [ #4627](https://github.com/gammapy/gammapy/issues/4627) 
The initial PR is to add the `energy_edges` to the doc string in `ASmoothMapEstimator`

Documentation for the `energy_edges` will be added in due time.

### CODE REVIEW

File: gammapy/estimators/map/asmooth.py [hunk 1]
```diff
@@ -42,6 +42,8 @@ class ASmoothMapEstimator(Estimator):
         Significance estimation method.
     threshold : float
         Significance threshold.
+    energy_edges : `~astropy.units.Quantity`
+        Energy edges of the target maps bins.
```
Comment (bkhelifi) [inline gammapy/estimators/map/asmooth.py:46]: in this docstring, the edges is only 1 quantity, but in fact this is a list with [emin, emax]
Also, one of the user's issue is that the user's edge can not used, but rather the closest bins of the dataset map. This is the tricky point to explain

Comment (Astro-Kirsty) [reply gammapy/estimators/map/asmooth.py:46]: Yes for sure. 
@cgalelli and I are working on that together, our thought was to explain in the documentation what the `energy_edges` is, and then link to that in the doc string. 
We wanted to start a PR so we could see how committing to the same PR works.

**Comment by Astro-Kirsty** [comment]
Myself and @cgalelli have created an explanation for the energy_edges in the estimators section. 

We are open to comments on both the wording and the placement. Thanks in advance!

**Review by bkhelifi — APPROVED** [review]
Thanks a lot @Astro-Kirsty

File: docs/user-guide/estimators.rst [hunk 2]
```diff
@@ -37,6 +37,16 @@ In general the flux can be estimated using two methods:
    **backward folding**. This method is currently only exposed in the `ExcessMapEstimator`
 
 
+Energy edges
+---------------
```
Comment (registerrier) [inline docs/user-guide/estimators.rst:41]:
```suggestion
Energy edges
------------
```

File: docs/user-guide/estimators.rst [hunk 3]
```diff
@@ -37,6 +37,16 @@ In general the flux can be estimated using two methods:
...
+
+The input energy edges by the user are converted to the nearest parent dataset energy bin
+values. The estimators select the energy bins from the parent dataset which are closest to the
+requested energy edges. Therefore, the input energy edges are not always the same as the output
+energy bins provided in the final product. If a specific energy binning is required at the estimator
+level, it should be implemented in true_energy during the dataset computation.
```
Comment (registerrier) [inline docs/user-guide/estimators.rst:47]:
```suggestion
level, it should be implemented in the parent dataset geometry (i.e. the dataset energy axis edges should contain the required edges).
```

File: gammapy/estimators/map/asmooth.py [hunk 4]
```diff
@@ -42,6 +42,11 @@ class ASmoothMapEstimator(Estimator):
         Significance estimation method.
     threshold : float
         Significance threshold.
+    energy_edges : list of `~astropy.units.Quantity`
+        Edges of the target maps energy bins. The resulting bin edges will be the energy bins closest
+        to those of the parent dataset.
```
Comment (registerrier) [inline gammapy/estimators/map/asmooth.py:47]:
```suggestion
        Edges of the target maps energy bins. The resulting bin edges won't exactly be equal to the input ones but be 
        the closest values in the parent dataset energy axis edges.
```

File: docs/user-guide/estimators.rst (hunk 3 above)
Comment (registerrier) [reply docs/user-guide/estimators.rst:47]: Maybe one could say also that the estimators run on bins of reconstructed energy. They cannot modify the binning of the parent dataset but only group its bins. The requested edges will therefore be used to group the parent dataset energy edges into larger bins.

File: gammapy/estimators/map/asmooth.py [hunk 5]
```diff
@@ -42,6 +42,11 @@ class ASmoothMapEstimator(Estimator):
...
+        Default is None: obtain the edges directly from the parent dataset energy binning.
```
Comment (registerrier) [inline gammapy/estimators/map/asmooth.py:48]:
```suggestion
        Default is None: apply the estimator in each energy bin of the parent dataset.
```

**Review by registerrier — APPROVED** [review]
Thanks @Astro-Kirsty and @cgalelli . I have left a few inline suggestions.

**Review by bkhelifi — APPROVED** [review]
Thanks @Astro-Kirsty

**Review by registerrier — APPROVED** [review]
Thanks @Astro-Kirsty . No further comments.

**Pull request merged**